distributed/core.py

```diff
@@ -577,7 +575,7 @@ async def handle_stream(self, comm, extra=None, every_cycle=[]):
 
                 for func in every_cycle:
                     if is_coroutine_function(func):
-                        self.loop.add_callback(func)
```

Comment:
I tracked down the dangling task and this was the `ensure_compute` scheduled here. In particular it concerned the `ensure_computing` which uses the async interface to offload deserialisation. This was introduced in https://github.com/dask/distributed/pull/4307 just before the infamous `2020.12.0` release